src/main/java/org/folio/edge/orders/OrdersHandler.java

```diff
@@ -91,115 +88,107 @@ protected void handle(RoutingContext ctx, List<Routing> routingMapping) {
-        .setStatusCode(status);
-
-    if (xml != null) {
-      ctx.response()
-          .putHeader(HttpHeaders.CONTENT_TYPE, APPLICATION_XML)
-          .end(xml);
-    } else {
-      ctx.response()
-          .end();
-    }
-  }
-
   @Override
   protected void invalidApiKey(RoutingContext ctx, String key) {
     ResponseWrapper resp = new ResponseWrapper(
         new ErrorWrapper(ErrorCodes.API_KEY_INVALID.name(), MSG_INVALID_API_KEY + ": " + key));
-    handleError(ctx, 401, resp);
+    handleErrorResponse(ctx, 401, resp);
   }
 
   @Override
   protected void accessDenied(RoutingContext ctx, String msg) {
     ResponseWrapper resp = new ResponseWrapper(new ErrorWrapper(ErrorCodes.ACCESS_DENIED.name(), MSG_ACCESS_DENIED));
-    handleError(ctx, 401, resp);
+    handleErrorResponse(ctx, 401, resp);
   }
 
   @Override
   protected void badRequest(RoutingContext ctx, String body) {
     ResponseWrapper resp = new ResponseWrapper(new ErrorWrapper(ErrorCodes.BAD_REQUEST.name(), body));
-    handleError(ctx, 400, resp);
+    handleErrorResponse(ctx, 400, resp);
   }
 
   @Override
   protected void requestTimeout(RoutingContext ctx, String msg) {
     ResponseWrapper resp = new ResponseWrapper(
         new ErrorWrapper(ErrorCodes.REQUEST_TIMEOUT.name(), MSG_REQUEST_TIMEOUT));
-    handleError(ctx, 408, resp);
+    handleErrorResponse(ctx, 408, resp);
   }
 
   @Override
   protected void internalServerError(RoutingContext ctx, String msg) {
     Thread.dumpStack();
     if (!ctx.response().ended()) {
       ResponseWrapper resp = new ResponseWrapper(new ErrorWrapper(ErrorCodes.INTERNAL_SERVER_ERROR.name(), msg));
-      handleError(ctx, 500, resp);
+      handleErrorResponse(ctx, 500, resp);
+    }
+  }
+
+  private void handleResponseWithBody(RoutingContext ctx, HttpClientResponse response, String respBody) {
+    String contentType = response.headers().get(HttpHeaders.CONTENT_TYPE);
+    int status = response.statusCode();
+    if (isSuccessStatus(status)) {
+      ctx.response()
+        .putHeader(HttpHeaders.CONTENT_TYPE, contentType)
+        .setStatusCode(status)
+        .end(respBody);
+    } else {
+      processErrorResponse(ctx, respBody, contentType, status);
+    }
+  }
+
+  private void processErrorResponse(RoutingContext ctx, String respBody, String contentType, int status) {
+    String acceptHeader = ctx.request().getHeader(HttpHeaders.ACCEPT);
+    if (contentType.equals(acceptHeader)) {
+      ctx.response()
+        .putHeader(HttpHeaders.CONTENT_TYPE, acceptHeader)
+        .setStatusCode(status)
+        .end(respBody);
+    } else {
+      ErrorCodes errorCode = Optional.ofNullable(ErrorCodes.fromValue(status)).orElse(ErrorCodes.INTERNAL_SERVER_ERROR);
+      ResponseWrapper resp = new ResponseWrapper(new ErrorWrapper(errorCode.name(), respBody));
+      handleErrorResponse(ctx, status, resp);
+    }
+  }
+
+  private boolean isSuccessStatus(int status) {
+    return status == 200 || status == 201;
```

Comment:
A better check is anything in the 2XX range... You're likely to see 204, and maybe even 202s for instance...